lxd/images.go

```diff
@@ -644,6 +644,19 @@ func imageCreateInPool(d *Daemon, info *api.Image, storagePool string) error {
 }
 
 func imagesPost(d *Daemon, r *http.Request) response.Response {
+	trusted, _, _, _ := d.Authenticate(r)
+	if !trusted {
+		secret := r.Header.Get("X-LXD-secret")
+		fingerprint := r.Header.Get("X-LXD-fingerprint")
+
+		if secret != "" && fingerprint != "" {
+			valid := imageValidSecret(fingerprint, secret)
+			if !valid {
+				return response.Forbidden(nil)
+			}
+		}
+	}
```

Comment:
Hmm, this reads as very very wrong to me.

If I do an unauthenticated POST without setting any of the headers, I'll go in `!trusted`, then blow past the `secret != "" && fingerprint != ""` and continue as if I was authenticated...

Reply:
If either of the headers aren't set, we must immediately return Forbidden, then if both set, do the comparison and return Forbidden if failed.

Reply:
This is still horribly broken!

Reply:
Actually, partly fixed later but there's still room for improvement and we shouldn't have a commit introduce a glaring security issue to then have a later commit fix it. This opens the door to bad cherry-picks getting only the security issue, so needs rebasing.